server/internal/thirdparty/toolmetrics/client.go

```diff
@@ -0,0 +1,338 @@
+			attribute.String("sort_order", pagination.SortOrder()),
+		),
+	)
+	defer span.End()
+
+	startTime := time.Now()
+
+	query := listLogsQueryDesc
+	if pagination.SortOrder() == "ASC" {
+		query = listLogsQueryAsc
+	}
+
+	perPage := pagination.Limit() - 1 // Remove the +1 for actual page size
+	rows, err := c.Conn.Query(ctx, query, projectID,
+		tsStart,
+		tsEnd,
+		cursor,
+		pagination.Limit())
+	if err != nil {
+		span.RecordError(err)
+		span.SetStatus(codes.Error, "failed to query logs")
+		c.Logger.ErrorContext(ctx, "failed to query tool logs from ClickHouse",
+			attr.SlogError(err),
+			attr.SlogProjectID(projectID),
+			slog.String(attrTsStart, tsStart.Format(time.RFC3339)),
+			slog.String(attrTsEnd, tsEnd.Format(time.RFC3339)),
+		)
+		return nil, fmt.Errorf("query logs: %w", err)
+	}
+
+	defer func(rows driver.Rows, logger *slog.Logger) {
+		err = rows.Close()
+		if err != nil {
+			logger.ErrorContext(ctx, "failed to close rows", attr.SlogError(err))
+		}
+	}(rows, c.Logger)
+
+	var results []ToolHTTPRequest
+	for rows.Next() {
+		var log ToolHTTPRequest
+		if err = rows.ScanStruct(&log); err != nil {
+			span.RecordError(err)
+			span.SetStatus(codes.Error, "failed to scan row")
+			c.Logger.ErrorContext(ctx, "failed to scan row",
+				attr.SlogError(err),
+				attr.SlogProjectID(projectID),
+			)
+			return nil, fmt.Errorf("error scanning row: %w", err)
+		}
+		results = append(results, log)
+	}
+
+	if err = rows.Err(); err != nil {
+		span.RecordError(err)
+		span.SetStatus(codes.Error, "rows iteration error")
+		c.Logger.ErrorContext(ctx, "error iterating rows",
+			attr.SlogError(err),
+			attr.SlogProjectID(projectID),
+		)
+		return nil, fmt.Errorf("rows error: %w", err)
+	}
+
+	// Calculate pagination metadata
+	hasNextPage := len(results) > perPage
+
+	// Trim to actual page size if we fetched extra for detection
+	if hasNextPage {
+		results = results[:perPage]
+	}
+
+	var nextPageCursor *time.Time
+	if len(results) > 0 {
+		// Next cursor is the timestamp of the last record
+		if hasNextPage {
+			lastTs := results[len(results)-1].Ts
+			nextPageCursor = &lastTs
+		}
+	}
+
+	queryDuration := time.Since(startTime)
+	span.SetAttributes(
+		attribute.Int("result_count", len(results)),
+		attribute.Bool("has_next_page", hasNextPage),
+		attribute.Float64("query_duration_ms", float64(queryDuration.Milliseconds())),
+	)
+	span.SetStatus(codes.Ok, "")
+
+	c.Logger.InfoContext(ctx, "successfully listed tool logs",
+		attr.SlogProjectID(projectID),
+		slog.Int(attrResultCount, len(results)),
+		slog.Bool(attrHasNextPage, hasNextPage),
+		slog.Float64(attrQueryDurationMs, float64(queryDuration.Milliseconds())),
+	)
+
+	return &ListResult{
+		Logs: results,
+		Pagination: PaginationMetadata{
+			PerPage:        perPage,
+			HasNextPage:    hasNextPage,
+			NextPageCursor: nextPageCursor,
+		},
+	}, nil
+}
+
+func (c *ClickhouseClient) Log(ctx context.Context, log ToolHTTPRequest) error {
```

Comment:
Did we decide we were going to maybe trim down what we log to clickhouse initially. Not request bodies and response bodies